mlreco/models/grappa.py

```diff
@@ -200,13 +205,34 @@ def __init__(self, cfg, name='grappa', batch_col=0, coords_col=(1, 4)):
                                                               eps=softplus_and_shift,
                                                               logspace=logspace)
                 else:
-                    self.momentum_net = MomentumNet(node_output_feats, num_output=1, num_hidden=momentum_config.get('num_hidden', 128))
+                    self.momentum_net = MomentumNet(node_output_feats, 
+                                                    num_output=1, 
+                                                    num_hidden=momentum_config.get('num_hidden', 128))
 
         self.vertex_mlp = base_config.get('vertex_mlp', False)
+        print(self.vertex_mlp)
         if self.vertex_mlp:
             node_output_feats = cfg[name]['gnn_model'].get('node_output_feats', 64)
-            vertex_config = cfg[name].get('vertex_net', {})
-            self.vertex_net = MomentumNet(node_output_feats, num_output=5, num_hidden=vertex_config.get('num_hidden', 128))
+            vertex_config = cfg[name].get('vertex_net', {'name': 'momentum_net'})
+            self.use_vtx_input_features = vertex_config.get('use_vtx_input_features', False)
+            if vertex_config['name'] == 'momentum_net':
+                self.vertex_net = MomentumNet(node_output_feats, 
+                                              num_output=5, 
+                                              num_hidden=vertex_config.get('num_hidden', 64),
+                                              evidential=True) # Enforce positive outputs
```

Comment:
Does this mean evidential is enabled by default? does this change the previous default behavior of vertex prediction?

Reply:
So `evidential` here is a vestigial name for changing the final output logit layer to pass through a softplus activation, which was needed to force `MomentumNet`'s output to be strictly positive.

We can change this to `positive_outputs=True` or the like

Reply:
Okay, `positive_outputs` is probably a better name. My question remains: if it is set to `True` by default, it means the final output logit layer passes through a softplus activation by default. You say this was not the case before, correct? Does it mean the current weights everyone is using was trained with `evidential=False` ? I think if that is the case, it might throw off the vertex net predictions with current weights, right?

Reply:
I think it won't crash when loading weights (since softplus doesn't have trainable weights) but if you have a trained vertex net and want to evaluate, yes that could be a problem. However did we have a working vertex-net at any point in time? (I can't remember)

Reply:
Not worried about the vertex coordinate prediction since that never worked. But we use it for primary particle prediction, if I am not mistaken. (The first 2 scores of the logits returned by vertex net)